akka-typed-tests/src/test/scala/akka/typed/persistence/scaladsl/PersistentActorSpec.scala

```diff
@@ -140,6 +180,46 @@ class PersistentActorSpec extends TypedSpec(PersistentActorSpec.config) with Eve
       }
     }
 
+    /**
+     * Verify that all side-effects callbacks are called (in order) and only once.
+     * The [[IncrementTwiceAndThenLog]] command will emit two Increment events
+     */
+    def `chainable side effects with events`(): Unit = {
+      sideEffectLogging.reset()
+      val c = start(counter("c5"))
+
+      val probe = TestProbe[State]
+
+      c ! IncrementTwiceAndThenLog
+      c ! GetValue(probe.ref)
+      probe.expectMsg(State(2, Vector(0, 1)))
+      sideEffectLogging.logs shouldBe List(firstLogging, secondLogging)
```

Comment:
`sideEffectLogging` is used both from the test thread and the actor's thread, and it's not thread safe. Make it thread safe, `synchronized` is fine in tests.

Reply:
or use a TestProbe

Reply:
I'm creating the `SideEffectLogging` in scope on each test now. 

Curious to know how would you use a `TestProbe` to solve that. Do you mean injecting another actor to receive the side-effect messages and use the probe to inspect it?

Reply:
Let the counter take an `ActorRef` and send the side effect to that instead, pass `TestProbe.ref` to it and then you can assert with expectMessage and friends.

Reply:
Thanks! Pushed a new version using TestProbe.

Let me know if any other small improvement.